### ISSUE TITLE
FindAllMarkers returns different avg_log2FC values different Seurat versions

### ISSUE THREAD

**Issue by PietroD** [satijalab/seurat#7862]
FindAllMarkers returns different avg_log2FC values between Seurat version 4.0.3 and 4.3.0.

Cannot reproduce it on pbmc3k dataset though. 
Using in house data the differences are striking in avg_log2FC values and number of genes.

Here is the code on in house data:

```r
library('Seurat', lib.loc = "~/Documents/myRpackages/")
packageVersion('Seurat')
[1] ‘4.0.3’

# or 
# library('Seurat')
# packageVersion('Seurat')
# [1] ‘4.3.0’

data <-
  readRDS('dataset.rds')
  
Idents(data) <- 'seurat_clusters'

markers <- FindAllMarkers(data, 
                          assay = 'RNA', 
                          logfc.threshold = 0.25,
                          min.pct = 0.1, 
                          only.pos = T, 
                          verbose = T)

```

Output from version 4.0.3:

<img width="1150" alt="Screenshot 2023-10-02 at 11 55 19" src="https://github.com/satijalab/seurat/assets/33426755/656495a5-e834-49a5-80f1-6b048579bebf">

Output from version 4.3.0:

<img width="1144" alt="Screenshot 2023-10-02 at 11 56 27" src="https://github.com/satijalab/seurat/assets/33426755/e065142f-8b95-4128-a450-2be992504e22">

```r
R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.6.3

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] SeuratObject_4.1.3 Seurat_4.3.0      

loaded via a namespace (and not attached):
  [1] nlme_3.1-162           matrixStats_0.63.0     spatstat.sparse_3.0-1 
  [4] RcppAnnoy_0.0.20       RColorBrewer_1.1-3     httr_1.4.5            
  [7] sctransform_0.3.5      tools_4.2.3            utf8_1.2.3            
 [10] R6_2.5.1               irlba_2.3.5.1          KernSmooth_2.23-20    
 [13] DBI_1.1.3              uwot_0.1.14            lazyeval_0.2.2        
 [16] colorspace_2.1-0       sp_1.6-0               tidyselect_1.2.0      
 [19] gridExtra_2.3          compiler_4.2.3         progressr_0.13.0      
 [22] cli_3.6.0              spatstat.explore_3.1-0 plotly_4.10.1         
 [25] scales_1.2.1           lmtest_0.9-40          spatstat.data_3.0-1   
 [28] ggridges_0.5.4         pbapply_1.7-0          goftest_1.2-3         
 [31] stringr_1.5.0          digest_0.6.31          spatstat.utils_3.0-2  
 [34] pkgconfig_2.0.3        htmltools_0.5.4        parallelly_1.34.0     
 [37] fastmap_1.1.1          htmlwidgets_1.6.2      rlang_1.1.0           
 [40] rstudioapi_0.14        shiny_1.7.4            generics_0.1.3        
 [43] zoo_1.8-12             jsonlite_1.8.4         spatstat.random_3.1-4 
 [46] ica_1.0-3              dplyr_1.1.2.9000       magrittr_2.0.3        
 [49] patchwork_1.1.2        Matrix_1.5-3           Rcpp_1.0.10           
 [52] munsell_0.5.0          fansi_1.0.4            abind_1.4-5           
 [55] reticulate_1.28        lifecycle_1.0.3        stringi_1.7.12        
 [58] MASS_7.3-58.2          Rtsne_0.16             plyr_1.8.8            
 [61] grid_4.2.3             parallel_4.2.3         listenv_0.9.0         
 [64] promises_1.2.0.1       ggrepel_0.9.3          deldir_1.0-6          
 [67] miniUI_0.1.1.1         lattice_0.20-45        cowplot_1.1.1         
 [70] splines_4.2.3          tensor_1.5             pillar_1.9.0          
 [73] igraph_1.4.1           spatstat.geom_3.1-0    future.apply_1.10.0   
 [76] reshape2_1.4.4         codetools_0.2-19       leiden_0.4.3          
 [79] glue_1.6.2             data.table_1.14.8      png_0.1-8             
 [82] vctrs_0.6.0            httpuv_1.6.9           polyclip_1.10-4       
 [85] gtable_0.3.2           RANN_2.6.1             purrr_1.0.1           
 [88] tidyr_1.3.0            scattermore_0.8        future_1.32.0         
 [91] ggplot2_3.4.1          mime_0.12              xtable_1.8-4          
 [94] later_1.3.0            survival_3.5-3         viridisLite_0.4.1     
 [97] tibble_3.2.1           writexl_1.4.2          cluster_2.1.4         
[100] globals_0.16.2         fitdistrplus_1.1-8     ellipsis_0.3.2        
[103] ROCR_1.0-11 
```

**Comment by yjchen1201** [comment]
I had a similar issue when i was using FindMarkers of Seurat (4.0.3 and 4.3.0) for my scATAC peak object. Seurat 4.3.0 gave me more differential regions and most avg_log2FC value went way higher than what 4.0.3 gave me

**Comment by PietroD** [comment]
> I had a similar issue when i was using FindMarkers of Seurat (4.0.3 and 4.3.0) for my scATAC peak object. Seurat 4.3.0 gave me more differential regions and most avg_log2FC value went way higher than what 4.0.3 gave me

In my case it was the other way around, but still a striking difference for which I could not find explanation in the changelog

**Comment by yjchen1201** [comment]
> > I had a similar issue when i was using FindMarkers of Seurat (4.0.3 and 4.3.0) for my scATAC peak object. Seurat 4.3.0 gave me more differential regions and most avg_log2FC value went way higher than what 4.0.3 gave me
> 
> In my case it was the other way around, but still a striking difference for which I could not find explanation in the changelog

Here could be the answers?
https://github.com/stuart-lab/signac/issues/1174
https://github.com/satijalab/seurat/issues/6915
https://github.com/satijalab/seurat/issues/7749

**Comment by saketkc** [comment]
I would recommend updating to Seurat 4.4.0 (or Seurat 5.0 once it is approved by CRAN) as there were some bug fixes in 4.4.0 on how fold change was calculated (and new fold change calculations in 5.0) that would address your concerns.
